attestation-agent/attester/src/tdx/mod.rs

```diff
@@ -9,10 +9,24 @@ use serde::{Deserialize, Serialize};
 use std::path::Path;
 use tdx_attest_rs;
 
+mod az;
+
 const CCEL_PATH: &str = "/sys/firmware/acpi/tables/data/CCEL";
 
 pub fn detect_platform() -> bool {
-    Path::new("/dev/tdx-attest").exists() || Path::new("/dev/tdx-guest").exists()
+    // Upstream kernel exposes a /dev/tdx_guest device but Intel's lib expects
+    // /dev/tdx-guest
+    let paths = [
+        Path::new("/dev/tdx-attest"),
+        Path::new("/dev/tdx-guest"),
+        Path::new("/dev/tdx_guest"),
+    ];
+    for path in paths {
+        if path.exists() {
+            return true;
+        }
+    }
+    false
```

Comment:
There is a rust way I have ever read. Just fYI : )
```rust
pub fn detect_platform() -> bool {
    ["/dev/tdx-attest", "/dev/tdx-guest", "/dev/tdx_guest"]
        .iter()
        .any(|p| Path::new(p).exists())
}
```

Reply:
thanks for the hint.

i've reworked this part to use cpuid like Mikko suggested. Can you look and tell me what you think about it?